circle.yml

```diff
@@ -0,0 +1,14 @@
+dependencies:
+  cache_directories:
+    - "~/.stack"
+  pre:
```

Comment:
The steps run every time. The ones here in `pre` take almost no time so I didn't think it was super important to put their artifacts into `~/.stack` (or some other cached directory). The `stack setup` and `stack build` steps are the time-consuming ones and their artifacts are what are cached via `~/.stack`.

You can see example timings and outcomes [here](https://circleci.com/gh/pbrisbin/tee-io/79). For example, the `stack build` step only compiles the actual application code, all dependencies were pre-installed because they live in `~/.stack` (presumably).